src/main/scala/latis/input/HapiAdapter.scala

```
@@ -4,40 +4,103 @@ import latis.model._
   
-  def apply(uri: URI): SampledFunction = 
-    HapiFunction (
-      model,
-      TextAdapter(model, config),
-      uri,
-      config.query
-    )
+  /**
+   * Defines the Adapter that will be used to parse the
+   * results from the HAPI service call
+   */
+  def parsingAdapter: Adapter
+  
+  /**
+   * Defines the requested data format.
+   * This must be consistent with the parsingAdapter.
+   */
+  def datasetFormat: String
+
+    
+
+  override def getData(baseUri: URI, ops: Seq[Operation] = Seq.empty): SampledFunction = {
+    val query = buildQuery(ops)
+    val uri = new URI(baseUri.toString + "data?" + query) //TODO: make sure baseURI has separator
+    parsingAdapter.getData(uri)
+  }
+    
+  override def canHandleOperation(op: Operation): Boolean = op match {
+    case Selection("time",_,_) => true
+    //case p: Projection => true
+    case _ => false
+  }
+  
+  def buildQuery(ops: Seq[Operation]): String = {
+    // Define the format of time strings used by the HAPI API.
+    //val timeFormat = TimeFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
+    
+    // Placeholder for min time
+    var startTime: Option[String] = None
+    
+    // Placeholder for max time
+    var endTime:   Option[String] = None
+    
+    // Build the query string from the operations
+    val qs = ops map {
+      case Selection("time", op, value) =>
+        val time = value //TODO: timeFormat.format(value)
+        op.head match {
+          case '>' => "time.min=" + time
+          case '<' => "time.max=" + time
+          case '=' => ??? //TODO: support single time selection: add epsilon, take first
+          case _ => 
+            val msg = s"Unsupported select operator: $op"
+            throw new UnsupportedOperationException(msg)
+        }
+      //case Projection(vids @ _*) => vids.mkString("parameters=", ",", "")
+      case _ => ??? //TODO: bug, we should only be given Ops that pass canHandleOperation
```

Should we throw an exception here?